pallets/erc20-xcm-bridge/src/lib.rs

```diff
@@ -66,9 +66,10 @@ pub mod pallet {
 			Erc20Matcher::<T::Erc20MultilocationPrefix>::is_erc20_asset(asset)
 		}
 		pub fn weight_of_erc20_transfer() -> Weight {
+			let gas_limit = T::Erc20TransferGasLimit::get();
 			Weight::from_parts(
 				T::Erc20TransferGasLimit::get().saturating_mul(T::WeightPerGas::get().ref_time()),
-				0,
+				gas_limit / 4, // TODO: apply gas/proof_size ratio
```

Comment:
Not related to this PR but could erc20-xcm-bridge not be a pallet? it has no events, no storage and no extrinsic AFAICT. What is the purpose of making it a pallet?

Reply:
Oh ok, this could have done with some other junction though, maybe worth giving it a thought